src/SideNav.tsx

```diff
@@ -23,7 +30,7 @@ function SideNavBase({variant, className, bordered, children, ...props}) {
   )
 }
 
-const SideNav = styled(SideNavBase)`
+const SideNav = styled(SideNavBase)<StyledSideNavBaseProps>`
```

Comment:
I think styled-components can infer the type of the component being passed in:
```suggestion
const SideNav = styled(SideNavBase)`
```